### PR TITLE
Email service provider

### LINKED ISSUE #148: Email Service Provider
add toggle in UI (Host/site)
save/bind host level smtp settings

### PR DESCRIPTION
**Pull request by ashishpachauriap**
Close #148

### CODE REVIEW

File: DesktopModules/Vanjaro/UXManager/Extensions/Menu/YouTube/Controllers/SettingController.cs [hunk 1]
```diff
@@ -17,33 +18,51 @@ public class SettingController : UIEngineController
...
         public bool Save(dynamic Data)
         {
-            if (Core.Providers.Youtube.IsValid(Data.ToString()))
+            if (bool.Parse(Data.ApplyTo.ToString()))
             {
-                PortalController.UpdateEncryptedString(PortalSettings.PortalId, "Vanjaro.Integration.YouTube", Data.ToString(), Config.GetDecryptionkey());
-                return true;
+                if (Core.Providers.Youtube.IsValid(Data.Host_ApiKey.ToString()))
+                {
+                    PortalController.UpdatePortalSetting(PortalSettings.PortalId, "Vanjaro.Integration.YouTube.ApplyTo", "h");
```
Comment (nikhilmandeeps) [inline DesktopModules/Vanjaro/UXManager/Extensions/Menu/YouTube/Controllers/SettingController.cs]: please remove this settings value which are save in portal settings

**Review by nikhilmandeeps — CHANGES_REQUESTED** [review]

File: DesktopModules/Vanjaro/UXManager/Extensions/Menu/YouTube/Controllers/SettingController.cs (hunk 1 above)
Comment (ashishpachauriap) [reply DesktopModules/Vanjaro/UXManager/Extensions/Menu/YouTube/Controllers/SettingController.cs]: ApplyTo setting is removed from PortalSettings

File: DesktopModules/Vanjaro/UXManager/Extensions/Menu/YouTube/Controllers/SettingController.cs [hunk 2]
```diff
@@ -17,33 +18,49 @@ public class SettingController : UIEngineController
...
-        [HttpGet]
-        public string Delete()
+        [HttpPost]
+        public string Delete(dynamic Data)
         {
-            PortalController.UpdateEncryptedString(PortalSettings.PortalId, "Vanjaro.Integration.YouTube", string.Empty, Config.GetDecryptionkey());
+            if (bool.Parse(Data.ToString()))
+                HostController.Instance.UpdateEncryptedString("Vanjaro.Integration.YouTube.ApiKey", string.Empty, Config.GetDecryptionkey());
+            else
+                PortalController.UpdateEncryptedString(PortalSettings.PortalId, "Vanjaro.Integration.YouTube.ApiKey", string.Empty, Config.GetDecryptionkey());
```
Comment (nikhilmandeeps) [inline DesktopModules/Vanjaro/UXManager/Extensions/Menu/YouTube/Controllers/SettingController.cs]: why are you change the key name **Vanjaro.Integration.YouTube** to **this Vanjaro.Integration.YouTube.ApiKey**

File: DesktopModules/Vanjaro/UXManager/Extensions/Menu/Pixabay/Controllers/SettingController.cs [hunk 3]
```diff
@@ -17,33 +18,51 @@ public class SettingController : UIEngineController
...
         }
 
-        [HttpGet]
-        public string Delete()
+        [HttpPost]
+        public string Delete(dynamic Data)
         {
-            PortalController.UpdateEncryptedString(PortalSettings.PortalId, "Vanjaro.Integration.Pixabay", string.Empty, Config.GetDecryptionkey());
+            if (bool.Parse(Data.ToString()))
+                HostController.Instance.UpdateEncryptedString("Vanjaro.Integration.Pixabay.ApiKey", string.Empty, Config.GetDecryptionkey());
```
Comment (nikhilmandeeps) [inline DesktopModules/Vanjaro/UXManager/Extensions/Menu/Pixabay/Controllers/SettingController.cs]: why are you change the key name **Vanjaro.Integration.Pixabay** to this **Vanjaro.Integration.Pixabay.ApiKey**

**Review by nikhilmandeeps — CHANGES_REQUESTED** [review]

File: DesktopModules/Vanjaro/UXManager/Extensions/Menu/Pixabay/Controllers/SettingController.cs (hunk 3 above)
Comment (ashishpachauriap) [reply DesktopModules/Vanjaro/UXManager/Extensions/Menu/Pixabay/Controllers/SettingController.cs:63]: I believe this (Vanjaro.Integration.Pixabay.ApiKey) make more sense that what key we are going to save

File: DesktopModules/Vanjaro/UXManager/Extensions/Menu/YouTube/Controllers/SettingController.cs (hunk 2 above)
Comment (ashishpachauriap) [reply DesktopModules/Vanjaro/UXManager/Extensions/Menu/YouTube/Controllers/SettingController.cs:63]: I believe this (Vanjaro.Integration.YouTube.ApiKey) make more sense that what key we are going to save

Comment (ashishpachauriap) [reply DesktopModules/Vanjaro/UXManager/Extensions/Menu/YouTube/Controllers/SettingController.cs]: YouTube Key has been changed

File: DesktopModules/Vanjaro/UXManager/Extensions/Menu/Pixabay/Controllers/SettingController.cs (hunk 3 above)
Comment (ashishpachauriap) [reply DesktopModules/Vanjaro/UXManager/Extensions/Menu/Pixabay/Controllers/SettingController.cs]: Pixabay Key has been changed

**Review by nikhilmandeeps — APPROVED** [review]

**Pull request merged**